src/components/NewBook.js

```diff
@@ -1,39 +1,108 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import Card from './UI/Card';
+import { ADD_BOOK } from '../redux/books/books';
 
-const categories = ['Category', 'Action', 'Romance', 'Economy', 'History', 'Fantasy', 'Science Fiction'];
-
-const NewBook = () => (
-  <Card>
-    <h3>Add New Book</h3>
-    <form>
-      <div>
-        <label htmlFor="title">
-          Book Title:
-          <input type="text" id="title" name="title" placeholder="Book Title" />
-        </label>
-      </div>
-      <div>
-        <label htmlFor="author">
-          Author:
-          <input type="text" id="author" name="author" placeholder="Author" />
-        </label>
-      </div>
-      <div>
-        <label htmlFor="category">
-          Category:
-          <select id="category" name="category">
-            {categories.map((category) => (
-              <option key={category} value={category}>{category}</option>
-            ))}
+const NewBook = () => {
+  const [bookData, setBookData] = useState({
+    title: '',
+    isTitleValid: true,
+    author: '',
+    isAuthorValid: true,
+    category: 'Action',
+    isFormValid: false,
+  });
+
+  const {
+    title, isTitleValid, author, isAuthorValid, category, isFormValid,
+  } = bookData;
+  const categoryRef = useRef();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const isTitleValid = title.trim().length >= 3 && title.trim().length <= 55;
+    const isAuthorValid = author.trim().length >= 3 && author.trim().length <= 25;
+    const isFormValid = isTitleValid && isAuthorValid
+    && title.trim().length !== 0 && author.trim().length !== 0;
```

Comment:
- [OPTIONAL] I think you are trying to control the character of the input, you can simply do that by assigning the number to maxLength attribute which HTML5 provided us.